libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3Recognizer.cs

```diff
@@ -0,0 +1,510 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime;
+using Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime.Models;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Configuration;
+using Microsoft.Bot.Schema;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure.OData;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.AI.Luis
+{
+    /// <inheritdoc />
+    /// <summary>
+    /// A LUIS based implementation of <see cref="ITelemetryRecognizer"/> for the V3 endpoint.
+    /// </summary>
+    public class LuisV3Recognizer : ITelemetryRecognizer
+    {
+        /// <summary>
+        /// The value type for a LUIS trace activity.
+        /// </summary>
+        public const string LuisTraceType = "https://www.luis.ai/schemas/trace";
+
+        /// <summary>
+        /// The context label for a LUIS trace activity.
+        /// </summary>
+        public const string LuisTraceLabel = "LuisV3 Trace";
+
+        private readonly LuisApplication _application;
+        private readonly LuisV3PredictionOptions _options;
+        private readonly bool _includeApiResults;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LuisV3Recognizer"/> class.
+        /// </summary>
+        /// <param name="application">The LUIS application to use to recognize text.</param>
+        /// <param name="predictionOptions">(Optional) The LUIS prediction options to use.</param>
+        /// <param name="includeApiResults">(Optional) TRUE to include raw LUIS API response.</param>
+        /// <param name="clientHandler">(Optional) Custom handler for LUIS API calls to allow mocking.</param>
+        public LuisV3Recognizer(LuisApplication application, LuisV3PredictionOptions predictionOptions = null, bool includeApiResults = false, HttpClientHandler clientHandler = null)
+        {
```

Comment:
Strongly consider having this take an IHttpClientFactory, so that we can avoid the creation of the HttpClient on every call. 

[https://docs.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory?view=aspnetcore-2.2](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory?view=aspnetcore-2.2)

You can see some use of it here:
[https://github.com/microsoft/botbuilder-dotnet/search?q=GetHttpClient&unscoped_q=GetHttpClient](https://github.com/microsoft/botbuilder-dotnet/search?q=GetHttpClient&unscoped_q=GetHttpClient)

Reply:
I don't think we can use this here--we use .net standard 2.0 and it is listed as being asp.net core for 2.1 and 2.2.  @daveta can you chime in here and on the other open comments?

Reply:
I looked into this more and also talked to Dave.  There is no opportunity to use this here--and in the place where you pointed me, there is no benefit to what they are doing.
  if (customHttpClient != null)
            {
                var httpClientFactory = new ConstantHttpClientFactory(customHttpClient);
                this.authContext = new AuthenticationContext(configurationOAuth.Authority, true, new TokenCache(), httpClientFactory);
            }